main.scss

```
@@ -7,17 +7,30 @@
 @import 'src/scss/functions';
 @import 'src/scss/share';
 
-@if (not $o-share-is-silent) {
-	@include oShareBase;
+@mixin oShare($opts: (
+  'icons': $_o-share-icons,
+  'vertical': true,
+  'inverse': true
+)) {
+  $icons: map-get($opts, 'icons');
+  $vertical-enabled: map-get($opts, 'vertical');
+  $inverse-enabled: map-get($opts, 'inverse');
+
+  @include _oShareBase;
 
-	// run through all the different icons available
-	@each $icon in $_o-share-icons {
-		@include oShareActionIcon($icon);
-	}
+  @each $icon in $_o-share-icons {
+	  @include _oShareActionIcon($icon);
+  }
 
-	@include oShareInverse;
+  @include _oShareInverse;
 
-	.o-share--vertical {
-		@include oShareVertical();
-	}
+  @if $vertical-enabled {
+	  .o-share--vertical {
+		  @include _oShareVertical();
+	  }
+  }
+}
+
+@if (not $o-share-is-silent) {
+	@include oShare();
```

Our components normally reset silent mode so if the component is included again it's not output twice. Looks like o-share has never done this! 😬 
```suggestion
	@include oShare();
	// Set to silent again to avoid being output twice
	$o-share-is-silent: true !global;
```